### PR TITLE
build(ci): Remove travis builds and add local action to handle test artifacts

### PR DESCRIPTION
**Pull request by billyvg**
GHA builds have been pretty stable so far, so let's remove travis builds (except lint as there is a bug).
Adds codecov coverage artifact uploads to GHA builds.

### CODE REVIEW

File: .github/actions/artifacts/action.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: 'Sentry Artifacts'
+description: 'Handles uploading coverage/test artifacts to codecov and zeus'
```
Comment (joshuarli) [inline .github/actions/artifacts/action.yml:2]: No zeus?

Comment (joshuarli) [reply .github/actions/artifacts/action.yml:2]: Just don't know if the intention here is to remove zeus, or keep it (and it was not ported over)

Comment (billyvg) [reply .github/actions/artifacts/action.yml:2]: Oops yeah, no zeus.

**Review by joshuarli — APPROVED** [review]
simpsonsburnvillain

**Pull request merged**